addons/pos_paytm/models/pos_payment_method.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+import requests
+
+from odoo.addons.pos_paytm import utils as paytm_utils
+from odoo.exceptions import UserError
+from odoo import fields, models, api, _
+from datetime import datetime
+from dateutil import tz
+
+_logger = logging.getLogger(__name__)
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    paytm_mid = fields.Char(string='PayTM Merchant ID')
+    paytm_tid = fields.Char(string='PayTM Terminal ID')
+    merchant_key = fields.Char(string='PayTM Merchant Key')
+    channel_id = fields.Char(string='PayTM Channel ID', default='EDC')
+    paytm_test_mode = fields.Boolean(string='Test Mode', help='Run transactions in the test environment.')
+    accept_payment = fields.Selection(selection=[('auto', 'Automatically'), ('manual', 'Manually')], default='auto')
+    allowed_payment_modes = fields.Selection(selection=[('all', 'All'), ('card', 'Card'), ('qr', 'QR')], default='all')
+
+    def _get_payment_terminal_selection(self):
+        selection = super()._get_payment_terminal_selection()
+        selection += [('paytm', 'PayTM')]
+        return selection
+
+    def _paytm_make_request(self, url, payload=None):
+        """ Make a request to PayTM API.
+
+        Note: self.ensure_one()
+
+        :param str url: The url to be reached by the request.
+        :param dict payload: The payload of the request.
+        :return The JSON-formatted content of the response.
+        :rtype: dict
+        """
+        self.ensure_one()
+        try:
+            response = requests.post(url, json=payload)
+            response.raise_for_status()
+        except requests.exceptions.RequestException:
+            _logger.warning("Cannot connect with PayTM")
+            return {'body':{'resultInfo': {'resultCode': "F", "resultMsg": _("Unable to establish connection with PayTM.")}}}
```

Comment:
Instead, you should pass the server's response as a parameter.
The specific error. Because there we know that there is a connection problem but nothing more.
Off the payTM server certainly responds what's wrong;
- wrong currency
- bad credential
- account error
Or does not respond at all;
- error 500
- timeout